ztf_viewer/pages/viewer.py

```diff
@@ -775,6 +795,131 @@ def set_title(oid, dr):
+        Input("light-curve-type", "value"),
+        Input("fold-period", "value"),
+        Input("fold-zero-phase", "value"),
+        Input(dict(type="ref-mag-input", index=ALL), "id"),
+        Input(dict(type="ref-mag-input", index=ALL), "value"),
+        Input(dict(type="ref-magerr-input", index=ALL), "id"),
+        Input(dict(type="ref-magerr-input", index=ALL), "value"),
+        Input("additional-light-curves", "value"),
+        Input("webgl-is-available", "children"),
+        Input("models-fit-dd", "value"),
+    ],
+)
+def fit_lc(
+    cur_oid,
+    dr,
+    different_filter,
+    different_field,
+    min_mjd,
+    max_mjd,
+    lc_type,
+    period,
+    phase0,
+    ref_mag_ids,
+    ref_mag_values,
+    ref_magerr_ids,
+    ref_magerr_values,
+    additional_lc_types,
+    webgl_available,
+    name_model,
+):
+    if lc_type == "folded" and not period:
+        raise PreventUpdate
+
+    if min_mjd is not None and max_mjd is not None and min_mjd >= max_mjd:
+        raise PreventUpdate
+
+    ref_mag = immutabledefaultdict(
+        lambda: np.inf, {id["index"]: value for id, value in zip(ref_mag_ids, ref_mag_values) if value is not None}
+    )
+    ref_magerr = immutabledefaultdict(
+        float, {id["index"]: value for id, value in zip(ref_magerr_ids, ref_magerr_values) if value is not None}
+    )
+
+    external_data = immutabledict(
+        {value: immutabledict({"radius_arcsec": ADDITIONAL_LC_SEARCH_RADIUS_ARCSEC}) for value in additional_lc_types}
+    )
+
+    # It is "0" or "1" or None
+    webgl_available = True if webgl_available is None else bool(int(webgl_available))
+    render_mode = "auto" if webgl_available else "svg"
+
+    other_oids = neighbour_oids(different_filter, different_field)
+    if lc_type == "full":
+        lcs = get_plot_data(
+            cur_oid,
+            dr,
+            other_oids=other_oids,
+            min_mjd=min_mjd,
+            max_mjd=max_mjd,
+            ref_mag=ref_mag,
+            ref_magerr=ref_magerr,
+            external_data=external_data,
+        )
+    elif lc_type == "folded":
+        offset = -(phase0 or 0.0) * period
+        lcs = get_folded_plot_data(
+            cur_oid,
+            dr,
+            period=period,
+            offset=offset,
+            other_oids=other_oids,
+            min_mjd=min_mjd,
+            max_mjd=max_mjd,
+            ref_mag=ref_mag,
+            ref_magerr=ref_magerr,
+            external_data=external_data,
+        )
+    else:
+        raise ValueError(f"{lc_type = } is unknown")
+
+    lcs = list(chain.from_iterable(lcs.values()))
+    df = pd.DataFrame.from_records(lcs)
+    coord = find_ztf_oid.get_sky_coord(cur_oid, dr)
+    ebv = sfd.ebv(coord)
+    items = []
+    column_width = 0
+    params = {}
+    if name_model:
+        params = model_fit.fit(df, name_model, dr, ebv)
+        items = [f"**{k}**: {np.round(float(params[k]), 3) if k!='error' else params[k]}" for k in params.keys()]
```

Comment:
More compact and hopefully more readable:
```suggestion
        items = [f"**{k}**: {np.round(float(v), 3) if k!='error' else v}" for k, v in params.items()]
```